addons/account/models/account_payment_term.py

```diff
@@ -136,7 +136,7 @@ def _compute_terms(self, date_ref, currency, company, tax_amount, tax_amount_cur
             term_vals = {
                 'date': line._get_due_date(date_ref),
                 'has_discount': line.discount_percentage,
-                'discount_date': None,
+                'discount_date': False,
```

Comment:
I think this isn't needed if you don't have the `if isinstance(current_value, float):`
Why do you check for floats?

Reply:
I don't think so as the "conversion" code only apply to the values of the dictionary. Here `discount_date` is used in the key of the dictionary (which is itself a dictionary). So: `{ {"discount_date": ... , ...} : {...} }`
We could think of converting the dictionary key as well, but in my opinion it's not necessary and in the context of this PR having the `None` value does not really make sense as it is never used in Odoo